File: dracpond/src/main/java/cc/chordflower/dracpond/application/arguments/Arguments.java
package cc.chordflower.dracpond.application.arguments;

import com.beust.jcommander.Parameter;
import org.jetbrains.annotations.Contract;

import java.util.Objects;

@SuppressWarnings( "unused" )
public class Arguments {

  @Parameter ( names = { "-v", "--verbose" }, description = "Enables verbose output" )
  private boolean verbose = false;

  @Parameter ( names = { "-c", "--config" }, description = "The location of the configuration file" )
  private String config = "";

  @Contract( pure = true ) public Arguments( ) {
    super( );
  }

  @Contract( pure = true ) public Arguments( boolean verbose, String config ) {
    super( );
    this.verbose = verbose;
    this.config = config;
  }

  @Contract( value = "null -> false", pure = true ) @Override
  public boolean equals( Object obj ) {
    if( this == obj ) {
      return true;
    }
    if( !( obj instanceof Arguments ) ) {
      return false;
    }
    Arguments other = ( Arguments ) obj;
    return Objects.equals( this.config, other.config ) && ( this.verbose == other.verbose );
  }

  @Contract( pure = true ) public final String getConfig( ) {
    return this.config;
  }

  @Override
  public int hashCode( ) {
    return Objects.hash( this.config, this.verbose );
  }

  @Contract( pure = true )
  public final boolean isVerbose( ) {
    return this.verbose;
  }

  @Override
  public String toString( ) {
    return String.format( "{ \"verbose\"=%s, \"config\"=\"%s\"}", this.verbose, this.config );
  }

}

File: dracpond/src/main/kotlin/cc/chordflower/dracpond/application/arguments/Arguments.kt
package cc.chordflower.dracpond.application.arguments

import com.beust.jcommander.Parameter
import org.jetbrains.annotations.Contract
import java.util.*

class Arguments() {
  @get:Contract(pure = true)
  @Parameter(names = ["-v", "--verbose"], description = "Enables verbose output")
  var isVerbose = false

  @get:Contract(pure = true)
  @Parameter(names = ["-c", "--config"], description = "The location of the configuration file")
  var config = ""

  constructor(verbose: Boolean, config: String) : this() {
    isVerbose = verbose
    this.config = config
  }

  @Contract(value = "null -> false", pure = true)
  override fun equals(other: Any?): Boolean {
    if(this === other) {
      return true
    }
    if(other !is Arguments) {
      return false
    }
    return config == other.config && isVerbose == other.isVerbose
  }

  override fun hashCode(): Int {
    return Objects.hash(config, isVerbose)
  }

  override fun toString(): String {
    return """{ "verbose"=$isVerbose, "config"="$config"}"""
  }
}
